src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
The info command may not return immediately.  For example when dependencies of the current file are out of date, the info command will only return after the dependencies have been processed.

As a result, doing a right click may hang emacs for some time (seconds, minutes) if we make this request synchronously.  I think we should just show the "find definition" menu entry unconditionally, and then do the info request asynchronously in the callback.

(A related problem occurs in auto-completion.  Since company-mode does not support async, auto-completion may hang...  That's one of the reasons why we don't do auto-completion on TAB by default.)